### PR TITLE
lightweight replacement for django-cronjobs

### LINKED ISSUE #5050: replace django-cronjobs with something better
part of #4997 

Current thinking is a lightweight wrapper `cron` command that just fires off a job from a settings defined list of possiblities.  

Most of the file locking is redundant because the jobs are some combination of a) trivial and quick b) create a celery task and exit c) execute rarely (once/twice a day).  There may be some jobs that need special consideration though.

### PR DESCRIPTION
**Pull request by eviljeff**
fixes #5050 
Doesn't provide any locking support though, so if we decide some of the jobs do require that, then those would need to be wrapped with `@olympia.files.utils.atomic_lock(...)`

### CODE REVIEW

File: src/olympia/amo/tests/test_commands.py [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+
+
+@override_settings(
+    CRON_JOBS={'sample_cron_job': 'olympia.amo.tests.test_commands'})
+@mock.patch('olympia.amo.tests.test_commands.sample_cron_job')
+def test_cron_command(_mock):
+    assert _mock.call_count == 0
+    call_command('cron', 'sample_cron_job', 'arg1', 'arg2')
+    assert _mock.call_count == 1
+    _mock.assert_called_with('arg1', 'arg2')
```
Comment (diox) [inline src/olympia/amo/tests/test_commands.py]: Also add tests both the 2 possible error cases to improve coverage (cron command called without args, cron command called with unknown cron)

File: src/olympia/amo/management/commands/cron.py [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+
+
+class Command(BaseCommand):
+    help = 'Run one of the predefined cron jobs'
+    args = '[name args...]'
+
+    def handle(self, *args, **opts):
+        if not args:
+            log.error("Cron called without args")
+            print 'These jobs are available:\n%s' % '\n'.join(
```
Comment (diox) [inline src/olympia/amo/management/commands/cron.py]: `self.stderr.write()` is preferred (or just `log.error` alone if the logging configuration is correct)

File: src/olympia/amo/management/commands/cron.py [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+                sorted(settings.CRON_JOBS.keys()))
+            sys.exit(1)
+
+        name, args = args[0], args[1:]
+        path = settings.CRON_JOBS.get(name)
+        if not path:
+            log.error("Cron called with an unknown cron job: %s %s" %
+                      (name, args))
+            print 'Unrecognized job name: %s' % name
```
Comment (diox) [inline src/olympia/amo/management/commands/cron.py]: As above, `self.stderr.write()` is better (and testable)

File: src/olympia/addons/tasks.py [hunk 4]
```diff
@@ -23,9 +23,6 @@
 from olympia.lib.es.utils import index_objects
 from olympia.versions.models import Version
 
-# pulling tasks from cron
-from . import cron  # noqa
```
Comment (diox) [inline src/olympia/addons/tasks.py]: I've been tracing the introduction of this import all the way from d0aa9d2915cf78761ebf53e4047e37d6f409d6b8, 7 years ago, and it doesn't make any sense... glad to see it gone :)

File: src/olympia/lib/settings_base.py [hunk 5]
```diff
@@ -1684,3 +1683,42 @@ def get_redis_settings(uri):
 WEBEXT_PERM_DESCRIPTIONS_LOCALISED_URL = (
     'https://hg.mozilla.org/l10n-central/{locale}/raw-file/tip/'
     'browser/chrome/browser/browser.properties')
+
+# List all jobs that should be callable with cron here.
```
Comment (diox) [inline src/olympia/lib/settings_base.py]: A quick comment to document what the key-values are would be nice

File: src/olympia/addons/tasks.py (hunk 4 above)
Comment (eviljeff) [inline src/olympia/addons/tasks.py]: I've no idea what it was even trying to do.

File: src/olympia/amo/tests/test_commands.py (hunk 1 above)
Comment (eviljeff) [inline src/olympia/amo/tests/test_commands.py]: https://github.com/mozilla/addons-server/pull/5068/commits/085e9d1c0fc4ab49963b09037f8f028c4dfea0c8

**Pull request merged**